Cython/Utility/ObjectHandling.c

```diff
@@ -1035,6 +1035,32 @@ static CYTHON_INLINE int __Pyx_PyDict_ContainsTF(PyObject* item, PyObject* dict,
     return unlikely(result < 0) ? result : (result == (eq == Py_EQ));
 }
 
+/////////////// PySetContains.proto ///////////////
+
+static int __Pyx_PySet_ContainsUnhashable(PyObject *set, PyObject *key) {
+    int result = -1;
+    if (PySet_Check(key) && PyErr_ExceptionMatches(PyExc_TypeError)) {
+        /* Convert key to frozenset */
+        PyObject *tmpkey;
+        PyErr_Clear();
+        tmpkey = PyFrozenSet_New(key);
+        if (tmpkey != NULL) {
+            result = PySet_Contains(set, tmpkey);
+            Py_DECREF(tmpkey);
+        }
+    }
+    return result;
+}
+
+static CYTHON_INLINE int __Pyx_PySet_ContainsTF(PyObject* key, PyObject* set, int eq) {
```

Comment:
Nice.

One little bit-pick: If you define the whole block in a `.proto` section like this, then it will end up before the user code in the generated `.c` file, next to the other pre-declarations. Please only define this signature prototype in that section to make it globally available, and move the implementation into a normal `PySetContains` section (without suffix), so that the helper function itself goes to a less user visible place at the end of the file.

Reply:
Uh, right, my mistake.